crates/solver-types/src/api.rs

```diff
@@ -16,71 +17,130 @@ pub struct AssetAmount {
-	#[serde(rename = "requestedMinOutputs")]
-	pub requested_min_outputs: Vec<AssetAmount>,
+	/// Requested outputs
+	#[serde(rename = "requestedOutputs")]
+	pub requested_outputs: Vec<RequestedOutput>,
 	/// Minimum quote validity duration in seconds
 	#[serde(rename = "minValidUntil")]
 	pub min_valid_until: Option<u64>,
 	/// User preference for optimization
 	pub preference: Option<QuotePreference>,
 }
 
-/// Quote optimization preferences.
+/// Quote optimization preferences following UII standard
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "kebab-case")]
 pub enum QuotePreference {
 	Price,
 	Speed,
 	InputPriority,
+	TrustMinimization,
+}
+
+/// EIP-712 compliant order structure
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct EIP712Order {
+	/// Signature type (eip-712 or erc-3009)
+	#[serde(rename = "signatureType")]
+	pub signature_type: SignatureType,
+	/// ERC-7930 interoperable address of the domain
+	pub domain: InteropAddress,
+	/// Primary type for EIP-712 signing
+	#[serde(rename = "primaryType")]
+	pub primary_type: String,
+	/// Message object to be signed and submitted
+	pub message: serde_json::Value,
 }
 
-/// Settlement order data for quotes.
+/// Supported signature types
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct SettlementOrder {
-	/// Settlement contract address
-	pub settler: String,
-	/// Settlement-specific data to be signed
-	pub data: serde_json::Value,
+#[serde(rename_all = "kebab-case")]
+pub enum SignatureType {
+	Eip712,
+	Erc3009,
+}
+
+/// Quote details matching the request structure
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct QuoteDetails {
+	/// Requested outputs for this quote
+	#[serde(rename = "requestedOutputs")]
+	pub requested_outputs: Vec<RequestedOutput>,
+	/// Available inputs for this quote
+	#[serde(rename = "availableInputs")]
+	pub available_inputs: Vec<AvailableInput>,
 }
 
-/// A quote option with all necessary execution details.
+/// A quote option following UII standard
 #[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct QuoteOption {
-	/// Settlement orders
-	pub orders: SettlementOrder,
-	/// Required token allowances
-	#[serde(rename = "requiredAllowances")]
-	pub required_allowances: Vec<AssetAmount>,
+pub struct Quote {
+	/// Array of EIP-712 compliant orders
+	pub orders: Vec<EIP712Order>,
+	/// Quote details matching request structure
+	pub details: QuoteDetails,
 	/// Quote validity timestamp
 	#[serde(rename = "validUntil")]
-	pub valid_until: u64,
+	pub valid_until: Option<u64>,
 	/// Estimated time to completion in seconds
-	pub eta: u64,
-	/// Total cost in USD
-	#[serde(rename = "totalFeeUsd")]
-	pub total_fee_usd: f64,
+	pub eta: Option<u64>,
 	/// Unique quote identifier
 	#[serde(rename = "quoteId")]
 	pub quote_id: String,
-	/// Settlement mechanism type
-	#[serde(rename = "settlementType")]
-	pub settlement_type: SettlementType,
+	/// Provider identifier
+	pub provider: String,
```

Comment:
Let's add a comment here that this is always hardcoded to oif-solver, and only relevant for aggregator or external entities to track where the quote came from?